### ISSUE TITLE
Help! In Performance scripts, could you please include end-to-end performance benchmark?

### ISSUE THREAD

**Issue by tjoymeed** [NVIDIA-NeMo/Megatron-Bridge#948]
Hi all,

I am using the https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/scripts/performance/configs/qwen3/qwen3_30b_a3b_llm_pretrain.yaml

perf_matrix:
  h100:
    num_gpus_16:
      common:
        num_gpus_per_node: 8
        seq_length: 4096
        gbs: 512
        cuda_graphs: false
        tp: 1
        pp: 2
        cp: 1
        vp: 12
        ep: 8
        etp: 1
        fsdp: false
      bf16:
        mbs: 1

My question is: other than throughput, have you measured end-to-end benchmark performance, e.g. MMLU, etc.

I am worried that optimizing the throughput alone might lead to worse benchmark performance in the end?

Thanks!

**Comment by tjoymeed** [comment]
i.e. lets say one config has loss 2.685 at the 2000th global step while another config has loss 2.5 at the 2000th global step. 

And the former has a higher throughput.

My question is: does the higher loss neccesarily and sufficiently mean it would have worse performance in MMLU and other standard benchmark performance?

**Comment by yaoyu-33** [comment]
Hi this is something we want to work on next. We will provide some scripts to show convergence results.

The perf scripts are mainly taking in pre-train scenarios, where you have constant sequence length. We usually treat training loss as the indicator of llm quality, but not 100% for sure. Any dataset you are using?

We can also provide MMLU eval as well.

**Comment by tjoymeed** [comment]
Thanks!

I would suggest to use public dataset for POC demo and comparions purposes. For example, a small sample from Nemotron-CC?

Btw, I see that you guys used the first 100 steps for measuring the throughputs.

But the first 100 steps are very volatile and transient. Maybe 1000-3000 steps are good measure for fair comparison across algorithms and configs?

**Comment by yaoyu-33** [comment]
Hi, internally we have longer runs. We will share a bit more results from our internal benchmark soon.

**Comment by sajadn** [comment]
@yaoyu-33 please update if you have additional information
